src/Features/CSharp/Portable/SignatureHelp/SignatureHelpUtilities.cs

```diff
@@ -83,5 +84,30 @@ internal static TextSpan GetSignatureHelpSpan(AttributeArgumentListSyntax argume
         {
             return CommonSignatureHelpUtilities.GetSignatureHelpSpan(argumentList, s_getAttributeArgumentListCloseToken);
         }
+
+        internal static bool IsTriggerParenOrComma<TSyntaxNode>(SyntaxToken token, Func<char, bool> isTriggerCharacter) where TSyntaxNode : SyntaxNode
+        {
+            // Don't dismiss if the user types ( to start a parenthesized expression or tuple
+            // Note that the tuple initially parses as a parenthesized expression 
+            if (token.IsKind(SyntaxKind.OpenParenToken) && token.Parent.IsKind(SyntaxKind.ParenthesizedExpression))
+            {
+                var parenthesizedExpr = ((ParenthesizedExpressionSyntax)token.Parent).WalkUpParentheses();
+                return parenthesizedExpr.Parent is ArgumentSyntax &&
```

Comment:
this is expression is too complex and confusing.  Can you break it out into simpler cases?